### ISSUE TITLE
Menu demo: items are deleted when selected

### ISSUE THREAD

**Issue by cbrewster** [servo/servo#11941]
This is a demo on the `about:newtab` on bhtml.
The checkbox to the left of the item should be checked rather than the item being deleted.
![menu_items_disappear](https://cloud.githubusercontent.com/assets/9086315/16473440/5adc362e-3e2a-11e6-96f0-89ad7c30c119.gif)

**Comment by ConnorGBrewster** [comment]
Adding P1 as I assume we want all the linked demos on the newtab page to work properly. Feel free to change it if it shouldn't be P1.

**Comment by glennw** [comment]
Unsurprisingly, but just to confirm - this occurs both with and without WR enabled.

**Comment by asajeffrey** [comment]
A minimal example which demonstrates this behaviour:
```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Minimal example for #11941</title>
    <style>
    ul li {
      color: #92a5bf;
      background-color: #f5f8fe;
      transition: linear 0.2s all;
    }
    ul.active li {
      background-color: red;
    }
    </style>
  </head>

  <body>

    <ul>
      <li>
	<input id='website' name='type' type='radio'>
	<label for='website'>Website</label>
      </li>
      <li>
	<input id='icon' name='type' type='radio'>
	<label for='icon'>Icon</label>
      </li>
      <li>
	<input id='mobile' name='type' type='radio'>
	<label for='mobile'>Mobile</label>
      </li>
    </ul>
  
  </body>
  <script>
    window.onload = function() {
      document.getElementsByTagName("ul")[0].setAttribute("class","active");
    };
  </script>
</html>
```

**Comment by asajeffrey** [comment]
OK, this is just weird:
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Minimal example for #11941</title>
    <style>
    ul.active li.unselected {
      color: red;
      transition: linear 0.2s all;
    }
    ul.active li.selected {
      color: blue;
      transition: linear 0.2s all;
    }
    </style>
  </head>

  <body>

    <ul>
      <li class="unselected">
	<input id='website' name='type' type='radio'>
	<label for='website'>Website</label>
      </li>
      <li>
	<input id='icon' name='type' type='radio'>
	<label for='icon'>Icon</label>
      </li>
      <li>
	<input id='mobile' name='type' type='radio'>
	<label for='mobile'>Mobile</label>
      </li>
    </ul>
  
  </body>
  <script>
    window.onload = function() {
	document.getElementsByTagName("ul")[0].setAttribute("class","active");
	var lis = document.getElementsByTagName("li");
	var selection = null;
	for (var i=0; i<lis.length; i++) {
	    lis[i].setAttribute("class","unselected");
	    lis[i].onclick = function() {
		if (selection) { selection.setAttribute("class","unselected"); }
		selection = this;
		this.setAttribute("class","selected");
	    }
	}			       
    };
  </script>
</html>
```
The bullet on the li ends up a different color than its text!

**Comment by mbrubeck** [comment]
Definitely an incremental layout bug; can’t reproduce it when running with `--nonincremental-layout`.

**Comment by asajeffrey** [comment]
For me, the 2nd example is weird even with `--nonincremental-layout`.

**Comment by asajeffrey** [comment]
IRC chat: http://logs.glob.uno/?c=mozilla%23servo&s=30+Jun+2016&e=30+Jun+2016#c467566

TL;DR: @metajack says "can you rpelace it with spinner?".

**Comment by asajeffrey** [comment]
If this isn't on the splash page, fixing it isn't P1.

**Comment by metajack** [comment]
I wonder if this is the same error as on mozilla.org/en-US/. Mousing over any of the tiles removes them permanently.

**Comment by asajeffrey** [comment]
Could be, it's caused by css transitions not leaving things in the right state. I can believe mozilla.org is using transitions the same way.

**Comment by nox** [comment]
Test case from https://github.com/servo/servo/issues/11941#issuecomment-229717743 still behaves weirdly.

**Comment by lumiscosity** [comment]
Both test cases seem to work correctly now, tested on Arch Linux x86_64, servo built against commit id 9d29017c0dc6eec4ca7cd868cc13056e0bc0c64a, compared against Firefox 141.0 and Epiphany 48.5.

**Comment by jdm** [comment]
Thanks for checking!
